### ISSUE TITLE
could not convert string to float: neo4j-kernel, version: 3.3.5,69f246fab87bc58a931d27cfa5ec5c35f21b95fa

### ISSUE THREAD

**Issue by dmitrievav** [DataDog/integrations-extras#188]
**Output of the [info page](https://docs.datadoghq.com/agent/faq/agent-commands/#agent-status-and-information)**

```
# curl -H 'Accept: application/json; charset=UTF-8' -H 'Authorization: redacted' http://localhost:7474/db/manage/server/version
{
  "edition" : "community",
  "version" : "3.3.5"
}

# datadog-agent status

Getting the status from the agent.

==============
Agent (v6.4.1)
==============

  Status date: 2018-08-09 20:18:43.761257 UTC
  Pid: 104633
  Python Version: 2.7.9
  Logs:
  Check Runners: 1
  Log Level: info

  Paths
  =====
    Config File: /etc/datadog-agent/datadog.yaml
    conf.d: /etc/datadog-agent/conf.d
    checks.d: /etc/datadog-agent/checks.d

  Clocks
  ======
    NTP offset: -0.000640376 s
    System UTC time: 2018-08-09 20:18:43.761257 UTC

  Host Info
  =========
    bootTime: 2018-08-06 22:30:37.000000 UTC
    kernelVersion: 3.16.0-4-amd64
    os: linux
    platform: debian
    platformFamily: debian
    platformVersion: 8.10
    procs: 205
    uptime: 251266

....

=========
Collector
=========

  Running Checks
  ==============

    neo4j (unversioned)
    -------------------
      Total Runs: 1
      Metric Samples: 0, Total: 0
      Events: 0, Total: 0
      Service Checks: 1, Total: 1
      Average Execution Time : 21ms
      Error: could not convert string to float: neo4j-kernel, version: 3.3.5,69f246fab87bc58a931d27cfa5ec5c35f21b95fa
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/checks/base.py", line 303, in run
          self.check(copy.deepcopy(self.instances[0]))
        File "/etc/datadog-agent/checks.d/neo4j.py", line 137, in check
          self.gauge(self.display.get(doc['row'][0].lower(),""), doc['row'][1], tags=tags)
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/checks/base.py", line 132, in gauge
          self._submit_metric(aggregator.GAUGE, name, value, tags=tags, hostname=hostname, device_name=device_name)
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/checks/base.py", line 129, in _submit_metric
          aggregator.submit_metric(self, self.check_id, mtype, name, float(value), tags, hostname)
      ValueError: could not convert string to float: neo4j-kernel, version: 3.3.5,69f246fab87bc58a931d27cfa5ec5c35f21b95fa
```

**Additional environment details (Operating System, Cloud provider, etc):**

I added `self.log.info("doc: %s" % (doc))` to see the original value we trying to convert:

```
2018-08-09 22:44:11 UTC | INFO | (datadog_agent.go:137 in LogMessage) | (neo4j.py:136) | doc: {u'meta': [None, None], u'row': [u'KernelVersion', u'neo4j-kernel, version: 3.3.5,*****
******************************b95fa']}
```

**Steps to reproduce the issue:**

Followed setup instructions from here:
https://docs.datadoghq.com/integrations/neo4j/

**Additional information you deem important (e.g. issue happens only occasionally):**

https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/checks/base.py (line 129):

`aggregator.submit_metric(self, self.check_id, mtype, ensure_bytes(name), float(value), tags, hostname)`

Trying converts metric to float, when originally it is string ...

To work around I disabled few problematic metrics:

```
--- /etc/datadog-agent/checks.d/neo4j.py  2018-08-08 09:53:14.712361122 +0000
+++ /etc/datadog-agent/checks.d/neo4j.py  2018-08-09 22:45:51.643572539 +0000
@@ -16,8 +16,8 @@

     # Neo4j metrics to send
     keys = set([
-        'kernelversion',
-        'storeid',
+        'off_kernelversion',
+        'off_storeid',
         'storecreationdate',
         'storelogversion',
         'kernelstarttime',
@@ -133,6 +133,7 @@
             self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=service_check_tags)

         for doc in stats['results'][0]['data']:
+            #self.log.info("doc: %s" % (doc))
             if doc['row'][0].lower() in self.keys:
                 self.gauge(self.display.get(doc['row'][0].lower(),""), doc['row'][1], tags=tags)
```

**Comment by dabcoder** [comment]
Ping for the maintainer @davidfauth.

**Comment by yzhan289** [comment]
Pinging @davidfauth, any updates for this?

**Comment by yzhan289** [comment]
Hi @dmitrievav, support for Neo4j 4.x has been released with integration version 2.0.0: https://github.com/DataDog/integrations-extras/pull/944. This will require Agent 7.34, which will be released within the next few weeks. Can you try updating to 2.0.0 once it is out? That may resolve your issues. I will close the Github issue in the meantime, but feel free to reopen if this is still an issue.
